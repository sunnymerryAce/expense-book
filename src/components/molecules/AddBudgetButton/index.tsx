import React, { useState } from "react";
import moment from "moment";
import Box from "@material-ui/core/Box";
import TextButton, { TextButtonProps } from "@/components/atoms/TextButton";
import MonthInput, { MonthInputProps } from "@/components/atoms/MonthInput";

export interface AddBudgetButtonProps {
  addBudget: (yyyymm: string) => void;
}

const AddBudgetButton: React.FC<AddBudgetButtonProps> = (props) => {
  const [isOpen, setIsOpen] = useState<boolean>(false);

  const addButtonProps: TextButtonProps = {
    text: "予算を追加する",
    handleClick: () => {
      setIsOpen(true);
    },
  };

  const monthInputProps: MonthInputProps = {
    isOpen,
    handleChange: (date: Date) => {
      props.addBudget(moment(date).format("YYYYMM"));
    },
    handleClose: () => {
      setIsOpen(false);
    },
  };

  return (
    <>
      <Box padding="20px 0">
        <TextButton {...addButtonProps}></TextButton>
      </Box>
      <MonthInput {...monthInputProps}></MonthInput>
    </>
  );
};

export default AddBudgetButton;
