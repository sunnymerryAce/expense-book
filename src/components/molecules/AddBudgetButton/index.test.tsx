import React from "react";
import AddBudgetButton from "./";
import { shallow, mount } from "enzyme";

describe("<AddBudgetButton />", () => {
  let container: any = null;

  const props = {
    addBudget: jest.fn(),
  };

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    container = shallow(<AddBudgetButton {...props} />);
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper props:TextButton", () => {
    container = shallow(<AddBudgetButton {...props} />);
    expect(container.find("TextButton").props()).toMatchObject({
      text: "予算を追加する",
    });
  });

  test("should show date picker on click", () => {
    container = mount(<AddBudgetButton {...props} />);
    container.find("TextButton").simulate("click");
    expect(container.find(".MuiPickersModal-dialogRoot").length).toBeTruthy();
  });
});
