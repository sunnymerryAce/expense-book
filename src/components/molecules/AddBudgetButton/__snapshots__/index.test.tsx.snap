// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBudgetButton /> should match the snapshot 1`] = `"<div class=\\"MuiBox-root MuiBox-root-1\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined TextButton makeStyles-root-2 MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">予算を追加する</span></button></div><div class=\\"MuiFormControl-root MuiTextField-root MonthInput-DatePicker makeStyles-datePicker-3\\" format=\\"yyyy/MM\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">年月</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" readonly=\\"\\" value=\\"2020/06\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div>"`;
